MUSICA linking fails on Derecho

### What happened?

Currently CAM-SIMA assumes it can link to the `musica` and `jsonfortran` libraries whenever one uses a compset other than `FPHYStest`.  However, if then on tries to build on Derecho it fails with the following error:

```
/usr/bin/ld: cannot find -lmusica
/usr/bin/ld: cannot find -ljsonfortran
collect2: error: ld returned 1 exit status
```

which indicates that the build system is currently unable to find or link to these libraries on Derecho.

Pinging @mattldawson @K20shores @boulderdaze  so that they know about this.

### What are the steps to reproduce the bug?

The following steps on Derecho will reproduce the error, assuming `<CAM-SIMA>` is wherever the CAM-SIMA codebase is located, with externals already checked out:

```
<CAM-SIMA>/cime/scripts/create_newcase --case cam_sima_test --res ne5pg2_ne5pg2_mg37 --mach derecho --compset FKESSLER --compiler gnu --run-unsupported
cd cam_sima_test
./case.setup
./case.build
```

### What CAM-SIMA hash were you using?

7d299b6

### What machine were you running CAM-SIMA on?

CISL machine (e.g. cheyenne)

### What compiler were you using?

GNU

### Path to a case directory, if applicable

/glade/derecho/scratch/nusbaume/model_runs/SIMA_ne5pg2_FKESSLER_test

### Will you be addressing this bug yourself?

No

### Extra info

If anyone does run into this bug, then it can be fixed by simply running:

`./xmlchange CAM_LINKED_LIBS=' '`

In their case directory, and then running `./case.build` again.

--- a/cime_config/config_component.xml
+++ b/cime_config/config_component.xml
@@ -307,8 +307,6 @@
     <valid_values></valid_values>
     <default_value> </default_value>
     <values match="last" modifier="additive">
-      <!-- Turn on chemistry if not using physics testbed -->
-      <value compset="^((?!_CAM%PHYSTEST).)*$">-lmusica -ljsonfortran</value>
       <!-- Add MPAS if using mpas grid and not using physics testbed -->
       <value compset="^((?!_CAM%PHYSTEST).)*$" grid="a%mpasa[0-9]+">-lmpas</value>
     </values>

--- a/cime_config/testdefs/testlist_cam.xml
+++ b/cime_config/testdefs/testlist_cam.xml
@@ -36,7 +36,7 @@
       <option name="wallclock">00:10:00</option>
       <option name="comment">GNU build test for Moist Held-Suarez physics</option>
     </options>
-  </tests>
+  </test>
 
   <!-- Derecho dycore tests -->
   <test compset="FKESSLER" grid="mpasa480_mpasa480" name="SMS_Ln2" testmods="cam/outfrq_kessler_mpas_derecho_nooutput/">

--- a/src/physics/utils/physics_data.F90
+++ b/src/physics/utils/physics_data.F90
@@ -68,6 +68,8 @@ CONTAINS
       if (find_input_name_idx >= 0) then
          constituent_index = find_input_name_idx
          is_constituent = .true.
+      else
+         find_input_name_idx = no_exist_idx
       end if
 
       !Loop through physics variable standard names:

